DOC: add ``np.linalg`` examples

luxedo opened the pull request:
<!--         ----------------------------------------------------------------
                MAKE SURE YOUR PR GETS THE ATTENTION IT DESERVES!
                ----------------------------------------------------------------

*  FORMAT IT RIGHT:
      https://www.numpy.org/devdocs/dev/development_workflow.html#writing-the-commit-message

*  IF IT'S A NEW FEATURE OR API CHANGE, TEST THE WATERS:
      https://www.numpy.org/devdocs/dev/development_workflow.html#get-the-mailing-list-s-opinion

*  HIT ALL THE GUIDELINES:
      https://numpy.org/devdocs/dev/index.html#guidelines

*  WHAT TO DO IF WE HAVEN'T GOTTEN BACK TO YOU:
      https://www.numpy.org/devdocs/dev/development_workflow.html#getting-your-pr-reviewed
-->
[skip actions][skip azp][skip cirrus]

Part of https://github.com/numpy/numpy/issues/21351.
Adds examples for `np.linalg` module: 

* `np.linalg.cross`: The example is an adaptation of `np.cross` examples.
* `np.linalg.diagonal`: The example is an adaptation of `np.diagonal` examples and adds a section to the `offset` argument.
* `np.linalg.matmul`: The example is an adaptation of `np.matmul` examples.
* `np.linalg.matrix_norm`: The example is an adaptation of `np.linalg.norm` examples.
* `np.linalg.outer`: The example is an adaptation of `np.outer` examples.
* `np.linalg.svdvals`: Completely new example.
* `np.linalg.trace`: The example is an adaptation of `np.trace` examples and adds a section to the `offset` argument
* `np.linalg.vecdot`: The example is an adaptation of `np.vecdot` examples.
* `np.linalg.vector_norm`: The example is an adaptation of `np.linalg.norm` examples.

The  examples for `np.linalg.matrix_transpose` already exists. This was a false positive from the script.

luxedo committed:
DOC: add `linalg.cross` example

[skip actions][skip azp][skip cirrus]
The example is an adaptation of `np.cross` examples.

luxedo committed:
DOC: add `linalg.diagonal` example

[skip actions][skip azp][skip cirrus]
The example is an adaptation of `np.diagonal` examples and adds a section to the `offset` argument.

luxedo committed:
DOC: add `linalg.matmul` example

[skip actions][skip azp][skip cirrus]
The example is an adaptation of `np.matmul` examples.

luxedo committed:
DOC: add `linalg.matrix_norm` example

[skip actions][skip azp][skip cirrus]
The example is an adaptation of `np.linalg.norm` examples.

luxedo committed:
DOC: add `linalg.outer` example

[skip actions][skip azp][skip cirrus]
The example is an adaptation of `np.outer` examples.

bmwoodruff commented:
> The examples for `np.linalg.matrix_transpose` already exists. This was a false positive from the script.

The code says to use the exact same docstring as the base function, and yep, the script gave a false positive for that reason. It doesn't check if the `__doc__` attribute has been manually set. Something to add. :)  If you find a way to test this, please share.
https://github.com/numpy/numpy/blob/787ff0f1ead66213a1bf18637057a98910ceb796/numpy/linalg/_linalg.py#L3246-L3251

This does make me wonder how much we want to maintain examples from two docstrings for almost the same function. I asked about it at the doc meeting last week, and they said to please add some, so we might as well.

luxedo committed:
DOC: add `linalg.svdvals` example

[skip actions][skip azp][skip cirrus]
Completely new example

luxedo committed:
DOC: add `linalg.trace` example

[skip actions][skip azp][skip cirrus]
The example is an adaptation of `np.trace` examples and adds a section to the `offset` argument.

luxedo committed:
DOC: add `linalg.vecdot` example

[skip actions][skip azp][skip cirrus]
The example is an adaptation of `np.vecdot` examples.

luxedo committed:
DOC: add `linalg.vector_norm` example

[skip actions][skip azp][skip cirrus]
The example is an adaptation of `np.vector_norm` examples.

mattip commented:
Thanks @luxedo
